src/components/lib/sl/sl.c

```diff
@@ -31,7 +32,8 @@ sl_cs_enter_contention(union sl_cs_intern *csi, union sl_cs_intern *cached, thdc
 		if (!ps_cas(&g->lock.u.v, cached->v, csi->v)) return 1;
 	}
 	/* Switch to the owner of the critical section, with inheritance using our tcap/priority */
-	if ((ret = cos_defswitch(csi->s.owner, t->prio, g->timeout_next, tok))) return ret;
+	if ((ret = cos_defswitch(csi->s.owner, t->prio, csi->s.owner == sl_thd_thdcap(g->sched_thd) ? 
```

Comment:
this is too clever.  Just do the `s.owner` initialization in a normal conditional.

Reply:
I don't understand the comment. You mean initialize `timeout`? Can you please elaborate?